### DISCUSSION TITLE
With the `--json` flag, how can I request that the "lines" section of the JSON is omitted?

### DISCUSSION THREAD

**Discussion by RecRanger** [BurntSushi/ripgrep#2821] [Q&A] [answered]
I'm searching very large binary files with sparse newlines, so some of the data that comes into the "lines" section is very very large.

Is there a way I can request that the "lines" section is omitted from the JSON response?

**Comment by BurntSushi** [comment, marked as answer]
Nope. You'll have to filter it out yourself. I'm not keen on adding different incompatible subsets of the JSON format.

I note that this generally shouldn't change the memory requirements. ripgrep already requires that each line, in full, is put on the heap. It does mean some extra deserialization/serialization work, but that should hopefully be pretty fast. And the additional memory requirements should be minimal if you do it one JSON line at a time.

**Reply by RecRanger** [reply]
Thank you!
